mitodl/open-discussions
# More stafflist index code

#### Pre-Flight checklist



#### What are the relevant tickets?
Related to #3893 

#### What's this PR do?
Adds some missing calls to update the stafflist index when appropriate

#### How should this be manually tested?
Similar to https://github.com/mitodl/open-discussions/pull/3893, but try these additional tests:
- Create a new public list, no items.  
- Go to http://localhost:8063/learn/search?type=stafflist and check the API search response in your network tab, the new list should not show up because it has no items
- Add an item to the list
- Wait a minute or two, search again, this time it should show up in API results
- Delete the item from the list
- Wait a bit, search again, this time it should not show up again because it's back to 0 items.
